src/blob/entry.rs

```diff
@@ -31,12 +32,15 @@ impl Entry {
             .read_exact_at_allocate(data_size + meta_size, self.header.meta_offset())
             .await
             .map_err(|err| err.into_bincode_if_unexpected_eof())
-            .context("Record load failed")?;
+            .with_context(|| format!("Record load failed from BLOB: {}", self.blob_file_name.to_path().display()))?;
         let mut buf = buf.freeze();
         let data_buf = buf.split_off(meta_size);
-        let meta = Meta::from_raw(&buf).map_err(|err| Error::from(err))?;
-        let record = Record::new(self.header.clone(), meta, data_buf);
-        record.validate()
+        let meta = Meta::from_raw(&buf)
+            .map_err(|err| Error::from(err))
+            .with_context(|| format!("Deserialization failed for Meta loaded from BLOB: {}", self.blob_file_name.to_path().display()))?;
```

Comment:
I think most of the error messages starts with lowercase in pearl

Reply:
I don't think it's critical. However, I've changed this